### PR TITLE
Remove slim from paddle framework

### PR DESCRIPTION
**Pull request by baiyfbupt**
### PR types

Others
### PR changes

Others
### Describe

Remove slim from paddle framework 

PaddleSlim所有功能已迁移到PaddleSlim repo
考虑到维护成本，后续正式的Paddle版本不再直接包含slim功能
保留量化底层代码在Paddle框架

### CODE REVIEW

File: python/setup.py.in [hunk 1]
```diff
@@ -164,14 +164,7 @@ packages=['paddle',
           'paddle.fluid.contrib.quantize',
           'paddle.fluid.contrib.reader',
           'paddle.fluid.contrib.slim',
-          'paddle.fluid.contrib.slim.core',
-          'paddle.fluid.contrib.slim.graph',
-          'paddle.fluid.contrib.slim.prune',
           'paddle.fluid.contrib.slim.quantization',
-          'paddle.fluid.contrib.slim.quantization.imperative',
```
Comment (qingqing01) [inline python/setup.py.in:171]: `paddle.fluid.contrib.slim.quantization.imperative`  is added recently, do not delete this.

Comment (baiyfbupt) [reply python/setup.py.in:171]: fixed

File: python/paddle/fluid/contrib/slim/tests/test_quantization_scale_pass.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ def conv_bn_layer(input,
             num_filters=ch_out,
             stride=stride,
             padding=padding,
+            use_cudnn=False,
```
Comment (wanghaoshuang) [inline python/paddle/fluid/contrib/slim/tests/test_quantization_scale_pass.py]: 这里为什么修改成False呢？

Comment (baiyfbupt) [reply python/paddle/fluid/contrib/slim/tests/test_quantization_scale_pass.py]: image
碰到了cudnn引起的单测fail

**Review by wanghaoshuang — DISMISSED** [review]

**Comment by baiyfbupt** [comment]
related to https://github.com/PaddlePaddle/Paddle/pull/25035

**Review by wanghaoshuang — DISMISSED** [review]

**Review by kolinwei — DISMISSED** [review]
PaddleSlim迁移到Slim repo，相关单测删除

File: python/paddle/fluid/contrib/slim/tests/test_graph.py [hunk 3]
```diff
@@ -1,146 +0,0 @@
-#   copyright (c) 2018 paddlepaddle authors. all rights reserved.
```
Comment (wzzju) [inline python/paddle/fluid/contrib/slim/tests/test_graph.py]: Hi, don't remove this unit test, thx.

**Review by wzzju — COMMENTED** [review]
Mostly LGTM.

File: python/paddle/fluid/contrib/slim/tests/test_graph.py (hunk 3 above)
Comment (baiyfbupt) [reply python/paddle/fluid/contrib/slim/tests/test_graph.py]: fixed

**Review by saxon-zh — APPROVED** [review]
LGTM

File: python/paddle/fluid/contrib/slim/__init__.py [hunk 4]
```diff
@@ -11,6 +11,3 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
```
Comment (wzzju) [inline python/paddle/fluid/contrib/slim/__init__.py:13]: The context of this file is empty. Can it be deleted  directly?

**Review by wzzju — APPROVED** [review]
LGTM.

**Review by kolinwei — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]
LGTM

**Pull request merged**